### PR TITLE
Improve demote - move broken packages to the end

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

File: lib/opam_solve.ml [hunk 1]
```diff
@@ -169,13 +169,19 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
                    (version, Ok opam_file))
 
   let demote_candidates_to_avoid versions =
-    let avoid_versions, regular_versions =
-      List.partition versions ~f:(fun (_version, opam_res) ->
+    let avoid_versions, broken_versions, regular_versions =
+      List.fold_left versions ~init:([], [], [])
+        ~f:(fun (avoid, broken, regular) ((_, opam_res) as e) ->
```
Comment (Leonidas-from-XIV) [inline lib/opam_solve.ml:174]: Could you reorder this to be `regular, avoid, broken` to match the final way they get appended?

**Review by Leonidas-from-XIV — APPROVED** [review]
Seems like a reasonable improvement to me, just a small nitpick from my side.

**Comment by samoht** [comment]
Updated with the suggestion

**Review by Leonidas-from-XIV — APPROVED** [review]

**Comment by Leonidas-from-XIV** [comment]
Perfect, thanks!

**Pull request merged**